MSYS2 compile warnings

This currently isn't an issue but I get the following warnings when compiling with GCC using msys2 ucrt64 on windows (this doesn't happen when compiling with MSVC or with GCC on linux)

```
[22/46] Building C object flac/src/libFLAC/CMakeFiles/FLAC.dir/metadata_iterators.c.obj
In file included from C:/project/flac/include/share/alloc.h:49,
                 from C:/project/flac/src/libFLAC/metadata_iterators.c:49:
In function 'transport_tempfile_',
    inlined from 'chain_rewrite_file_.constprop' at C:/project/flac/src/libFLAC/metadata_iterators.c:1539:7:
C:/project/flac/include/share/compat.h:191:21: warning: 'tempfilename' may be used uninitialized [-Wmaybe-uninitialized]
  191 | #define flac_rename rename_utf8
C:/project/flac/src/libFLAC/metadata_iterators.c:3568:17: note: in expansion of macro 'flac_rename'
 3568 |         if(0 != flac_rename(*tempfilename, filename)) {
      |                 ^~~~~~~~~~~
C:/project/flac/src/libFLAC/metadata_iterators.c: In function 'chain_rewrite_file_.constprop':
C:/project/flac/src/libFLAC/metadata_iterators.c:1481:15: note: 'tempfilename' was declared here
 1481 |         char *tempfilename;
      |               ^~~~~~~~~~~~
[32/46] Building C object flac/src/libFLAC/CMakeFiles/FLAC.dir/__/share/win_utf8_io/win_utf8_io.c.obj
C:/project/flac/src/share/win_utf8_io/win_utf8_io.c: In function 'get_utf8_argv':
C:/project/flac/src/share/win_utf8_io/win_utf8_io.c:132:29: warning: cast between incompatible function types from 'FARPROC' {aka 'long long int (*)()'} to 'int (*)(int *, wchar_t ***, wchar_t ***, int,  int *)' {aka 'int (*)(int *, short unsigned int ***, short unsigned int ***, int,  int *)'} [-Wcast-function-type]
  132 |         if ((wgetmainargs = (wgetmainargs_t)GetProcAddress(handle, "__wgetmainargs")) == NULL) {
      |                             ^
```

Also, cmake recently started throwing a deprecation warning when configuring

```
CMake Deprecation Warning at flac/CMakeLists.txt:5 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.
```

Thx for your quick work.
Tested the PR and there is still one annoying warning related to incompatible types

```
[32/46] Building C object flac/src/libFLAC/CMakeFiles/FLAC.dir/__/share/win_utf8_io/win_utf8_io.c.obj
C:/project/flac/src/share/win_utf8_io/win_utf8_io.c: In function 'get_utf8_argv':
C:/project/flac/src/share/win_utf8_io/win_utf8_io.c:132:29: warning: cast between incompatible function types from 'FARPROC' {aka 'long long int (*)()'} to 'int (*)(int *, wchar_t ***, wchar_t ***, int,  int *)' {aka 'int (*)(int *, short unsigned int ***, short unsigned int ***, int,  int *)'} [-Wcast-function-type]
  132 |         if ((wgetmainargs = (wgetmainargs_t)GetProcAddress(handle, "__wgetmainargs")) == NULL) {
      |                             ^
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,8 +1,4 @@
-# 3.1 is OK for most parts. However:
-# 3.3 is needed in src/libFLAC
-# 3.5 is needed in src/libFLAC/ia32
-# 3.9 is needed in 'doc' because of doxygen_add_docs()
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.12...3.31)
 
 if(NOT (CMAKE_BUILD_TYPE OR CMAKE_CONFIGURATION_TYPES OR DEFINED ENV{CFLAGS} OR DEFINED ENV{CXXFLAGS}))
     set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo")

--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -1,5 +1,3 @@
-cmake_minimum_required(VERSION 3.9)
-
 option(BUILD_DOXYGEN "Enable API documentation building via Doxygen if not already present" ON)
 
 if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/api")

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,3 @@
-cmake_minimum_required(VERSION 3.11)
-
 option(ENABLE_64_BIT_WORDS "Set FLAC__BYTES_PER_WORD to 8, for 64-bit machines. For 32-bit machines, turning this off might give a tiny speed improvement" ON)
 option(BUILD_UTILS "Build utils" OFF)
 

--- a/src/libFLAC/metadata_iterators.c
+++ b/src/libFLAC/metadata_iterators.c
@@ -1478,7 +1478,7 @@ static FLAC__bool chain_rewrite_metadata_in_place_(FLAC__Metadata_Chain *chain)
 static FLAC__bool chain_rewrite_file_(FLAC__Metadata_Chain *chain, const char *tempfile_path_prefix, const char *filename)
 {
 	FILE *f, *tempfile = NULL;
-	char *tempfilename;
+	char *tempfilename = 0;
 	FLAC__Metadata_SimpleIteratorStatus status;
 	const FLAC__Metadata_Node *node;
 

--- a/src/share/utf8/CMakeLists.txt
+++ b/src/share/utf8/CMakeLists.txt
@@ -1,5 +1,3 @@
-cmake_minimum_required(VERSION 3.12)
-
 add_library(utf8 STATIC
     charset.c
     iconvert.c

--- a/src/test_libFLAC++/metadata_manip.cpp
+++ b/src/test_libFLAC++/metadata_manip.cpp
@@ -342,7 +342,7 @@ static bool write_chain_(FLAC::Metadata::Chain &chain, bool use_padding, bool pr
 
 		if(to_different_file || chain.check_if_tempfile_needed(use_padding)) {
 			struct flac_stat_s stats;
-			FILE *file, *tempfile;
+			FILE *file, *tempfile = 0;
 			char *tempfilename;
 			if(preserve_file_stats) {
 				if(!get_file_stats_(filename, &stats))

